pkgs/by-name/et/etterna/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeDesktopItem,
+  copyDesktopItems,
+  # deps
+  clang,
+  cmake,
+  python2,
+  pkg-config,
+  openssl,
+  libGLU,
+  xorg,
+  alsa-lib,
+  libjack2,
+  libpulseaudio,
+  libogg,
+
+  # cfgs
+  enableCrashpad ? false,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "etterna";
+  version = "0.74.3";
+
+  src = fetchFromGitHub {
+    owner = "etternagame";
+    repo = "etterna";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-zzCk6axISswfTAk7rRha5HFzIHQ0AjpAZyAWzH+Cn1s=";
+  };
+
+  patches = [ ./fix-download-manager.patch ];
+
+  nativeBuildInputs = [
+    clang
+    cmake
+    pkg-config
+
+    copyDesktopItems
+  ] ++ lib.optionals enableCrashpad [ python2 ];
+
+  buildInputs = [
+    openssl
+    alsa-lib
+    libjack2
+    libpulseaudio
+    libGLU
+    libogg
+    xorg.libXinerama
+    xorg.libXrandr
+    xorg.libX11
+    xorg.libXext # Needed for DPMS
+    xorg.libXvMC
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "etterna";
+      desktopName = "Etterna";
+      genericName = "Rhythm and dance game";
+      icon = "etterna";
+      tryExec = "etterna";
+      exec = "etterna";
+      categories = [
+        "Application"
+        "Game"
+        "ArcadeGame"
+      ];
+      comment = "A cross-platform rhythm video game.";
+      terminal = false;
+    })
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/etterna}
+    mkdir -p $out/share/applications
+    # copy select necessary game files into virtual fs
+    cp -r /build/source/{Announcers,Assets,BGAnimations,BackgroundEffects,BackgroundTransitions,Data,GameTools,NoteSkins,Scripts,Themes} "$out/share/etterna/$dir"
+
+    # copy binary
+    cp /build/source/Etterna $out/bin/etterna-unwrapped
+
+    # Install the Icon
+    install -Dm644 /build/source/Docs/images/etterna-logo-light.svg "$out/share/icons/hicolor/scalable/apps/etterna.svg"
+
+    # wacky insertion of wrapper directly into phase, so that $out is set
+    cat > $out/bin/etterna << EOF
+    #!${stdenv.shell}
+    export ETTERNA_ROOT_DIR="\$HOME/.local/share/etterna"
+    export ETTERNA_ADDITIONAL_ROOT_DIRS="$out/share/etterna"
+    echo "HOME: \$HOME"
+    echo "PWD: \$(pwd)"
+    echo "ETTERNA_ADDITIONAL_ROOT_DIRS: \$ETTERNA_ADDITIONAL_ROOT_DIRS"
+    exec $out/bin/etterna-unwrapped "\$@"
+    EOF
+    chmod +x $out/bin/etterna
```

Comment:
Usually this would be done using makeWrapper / wrapProgram. In this case it would be tricky since you need the value of $HOME at runtime..
Also okay to keep it as is

Reply:
I'll keep this as is since I don't know how to translate that to `makeWrapper`/`wrapProgram`.

Reply:
I'm not sure about what you mean by "this workaround". Setting that environment variable seem to be intended. I also just checked to see if logging to a file was broken or not and that is true, it is broken.

Reply:
Didn't see the [issue](https://github.com/etternagame/etterna/issues/651) and Nix-related PR upstream until now. I'm fine this being merged as-is and might contribute to `etterna` later on my own time.